kgforge/core/commons/sparql_query_builder.py

```diff
@@ -345,20 +367,111 @@ def _replace_in_sparql(
         return qr
 
     @staticmethod
-    def apply_limit_and_offset_to_query(query, limit, default_limit, offset, default_offset):
+    def apply_limit_and_offset_to_query(
+        query, limit, default_limit, offset, default_offset
+    ):
         if limit:
             query = SPARQLQueryBuilder._replace_in_sparql(
-                query, "LIMIT", limit, default_limit,
-                re.compile(r" LIMIT \d+", flags=re.IGNORECASE)
+                query,
+                "LIMIT",
+                limit,
+                default_limit,
+                re.compile(r" LIMIT \d+", flags=re.IGNORECASE),
             )
         if offset:
             query = SPARQLQueryBuilder._replace_in_sparql(
-                query, "OFFSET", offset, default_offset,
-                re.compile(r" OFFSET \d+", flags=re.IGNORECASE)
+                query,
+                "OFFSET",
+                offset,
+                default_offset,
+                re.compile(r" OFFSET \d+", flags=re.IGNORECASE),
             )
 
         return query
 
+    @staticmethod
+    def create_select_query(
+        vars_: List[str],
+        statements: Union[str, List[str]],
+        distinct: bool,
+        search_in_graph: bool,
+        union: bool = False,
+    ):
+        if not union:
```

Comment:
Maybe you should check that statements is a string if not union? 
So a nested if and if it's not the case raise an Error that statements should be a string.

I would see it as an if that check union the boolean, and after that just an else case, and then within those, if statements.

if not union: (if not isinstance(statements, str) raise Ex, else ...)
else: (if not isinstance(statements, list) raise Ex, else ...)

Reply:
Done

Reply:
I still think the current if elif else structure is not very intuitive. It would be better to have the elif and else in a nested else

Reply:
Not sure to understand this. statements can be list or not when not union: no need for raising exception.

Reply:
if not union:
else: 
  if  isinstance(statements, list):
  else: 

is more intuitive